environment/src/main/java/com/sequenceiq/environment/experience/liftie/LiftieConnectorService.java

```diff
@@ -0,0 +1,112 @@
+
+import static com.sequenceiq.cloudbreak.util.NullUtil.putIfPresent;
+
+import java.net.URI;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.environment.experience.InvocationBuilderProvider;
+import com.sequenceiq.environment.experience.ResponseReader;
+import com.sequenceiq.environment.experience.RetryableWebTarget;
+import com.sequenceiq.environment.experience.api.LiftieApi;
+import com.sequenceiq.environment.experience.liftie.responses.DeleteClusterResponse;
+import com.sequenceiq.environment.experience.liftie.responses.ListClustersResponse;
+
+@Service
+public class LiftieConnectorService implements LiftieApi {
+
+    private static final String LIFTIE_CALL_EXEC_FAILED_MSG = "Something happened while the Liftie connection has attempted!";
+
+    private static final String LIFTIE_RESPONSE_RESOLVE_ERROR_MSG = "Unable to resolve Liftie response!";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LiftieConnectorService.class);
+
+    private final InvocationBuilderProvider invocationBuilderProvider;
+
+    private final RetryableWebTarget retryableWebTarget;
+
+    private final LiftiePathProvider liftiePathProvider;
+
+    private final ResponseReader responseReader;
+
+    private final Client client;
+
+    public LiftieConnectorService(LiftiePathProvider liftiePathProvider, LiftieResponseReader liftieResponseReader, RetryableWebTarget retryableWebTarget,
+            Client client, InvocationBuilderProvider invocationBuilderProvider) {
+        this.invocationBuilderProvider = invocationBuilderProvider;
+        this.liftiePathProvider = liftiePathProvider;
+        this.retryableWebTarget = retryableWebTarget;
+        this.responseReader = liftieResponseReader;
+        this.client = client;
+    }
+
+    @NotNull
+    @Override
+    public ListClustersResponse listClusters(@NotNull String env, @NotNull String tenant, @Nullable String workload, @Nullable Integer page) {
+        LOGGER.debug("About to connect Liftie API to list clusters");
+        WebTarget webTarget = client.target(liftiePathProvider.getPathToClustersEndpoint());
+        Map<String, String> queryParams = new LinkedHashMap<>();
+        putIfPresent(queryParams, "env", env);
+        putIfPresent(queryParams, "tenant", tenant);
+        putIfPresent(queryParams, "workloads", workload);
+        putIfPresent(queryParams, "page", page != null ? page.toString() : null);
+        webTarget = setQueryParams(webTarget, queryParams);
+        Invocation.Builder call = invocationBuilderProvider.createInvocationBuilder(webTarget);
```

Comment:
I would also add the invocation builder part into the try-catch.

Reply:
the try below is a try-with-resource but the invocationBuilderProvider.createInvocationBuilder is not AutoCloseable